collect_app/src/main/java/org/odk/collect/android/adapters/InstanceListCursorAdapter.java

```diff
@@ -57,26 +57,23 @@ public View getView(int position, View convertView, ViewGroup parent) {
         setUpSubtext(view);
 
         // Some form lists never contain disabled items; if so, we're done.
+        // Update: This only seems to be the case in Edit Saved Forms and it's not clear why...
         if (!shouldCheckDisabled) {
             return view;
         }
 
+        boolean formExists = true;
+        boolean isFormEncrypted = false;
+
         String formId = getCursor().getString(getCursor().getColumnIndex(InstanceColumns.JR_FORM_ID));
-        Cursor cursor = new FormsDao().getFormsCursorForFormId(formId);
+        String formVersion = getCursor().getString(getCursor().getColumnIndex(InstanceColumns.JR_VERSION));
+        Form form = new DatabaseFormsRepository().getLatestByFormIdAndVersion(formId, formVersion);
 
-        boolean formExists = false;
-        boolean isFormEncrypted = false;
-        if (cursor != null) {
-            try {
-                if (cursor.moveToFirst()) {
-                    int base64RSAPublicKeyColumnIndex = cursor.getColumnIndex(FormsColumns.BASE64_RSA_PUBLIC_KEY);
-                    String base64RSAPublicKey = cursor.getString(base64RSAPublicKeyColumnIndex);
-                    isFormEncrypted = base64RSAPublicKey != null && !base64RSAPublicKey.isEmpty();
-                    formExists = true;
-                }
-            } finally {
-                cursor.close();
-            }
+        if (form != null) {
+            String base64RSAPublicKey = form.getBASE64RSAPublicKey();
+            isFormEncrypted = base64RSAPublicKey != null && !base64RSAPublicKey.isEmpty();
```

Comment:
In some cases you check base64RSAPublicKey for null and length in other just for null. It would be good to be consistent.

Reply:
Yeah, that's probably me just copying logic from what was there before. There's no reason to check for an empty RSA key right?

Reply:
I would say no need to check for an empty RSA key.